shell/shared/shell.js

```diff
@@ -154,6 +166,49 @@ if (Meteor.isServer) {
     self.ready();
   });
 
+  Meteor.publish("myNotYetCompleteReferrals", function() {
+    if (!this.userId) {
+      return [];
+    }
+
+    // A not-yet-complete referral shows up as an Identity where referredBy=this.userId.
+    var notCompletedReferralIdentities = Meteor.users.find(
+      {referredBy: this.userId,
+       "profile.name": {$exists: true}},
+      {fields: {
+        "_id": 1,
+        "referredBy": 1,
+        "profile.name": 1}});
+    return notCompletedReferralIdentities;
+  });
+
+  Meteor.publish("myReferrals", function() {
+    if (!this.userId) {
+      return [];
+    }
+
+    // A completed referral shows up on ourselves, under Account.referredIdentityIds. So fetch the
+    // identity IDs, and make sure the client has that list of identity IDs. Separately we make sure
+    // the names names of the identities are sent, so the client can show useful information.
+    var completedIdentityIdsCursor = Meteor.users.find(
+      {_id: this.userId},
+      {fields: {"referredIdentityIds": 1}});
+    return completedIdentityIdsCursor;
+  });
+
+  Meteor.publish("myReferralsNames", function() {
+    var user = Meteor.users.findOne({_id: this.userId},
+                                    {fields: {"referredIdentityIds": 1}});
+    if (user && user.referredIdentityIds) {
+      // Completed referral IDs are not reactive at the moment.
+      return Meteor.users.find(
```

Comment:
I think that here you're revealing all of Bob's identities to Alice, not just the specific identity she referred. This could be bad for some of the user cases we had in mind for multiple identities. Perhaps this argues for keeping `referredBy` set on the identity, still referring to Alice (not a sentinel value), so that you can query for the exact identity that Alice actually referred.